openmdao/components/eq_constraint_comp.py

```diff
@@ -130,9 +130,10 @@ def compute(self, inputs, outputs):
             # scale factor that normalizes by the rhs, except near 0
             if options['normalize']:
                 # Indices where the rhs is near zero or not near zero
-                idxs_nz = np.where(cs_safe.abs(rhs) < 2)[0]
-                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)[0]
+                idxs_nz = np.where(cs_safe.abs(rhs) < 2)
+                idxs_nnz = np.where(cs_safe.abs(rhs) >= 2)
 
+                self._scale_factor = np.ones((rhs.shape))
```

Comment:
Same comment as above applies here.  Don't bother making `_scale_factor` an attribute if we're just using it locally and not saving it.